streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SavePointServiceImpl.java

```diff
@@ -171,6 +218,182 @@ public SavePoint getLatest(Long id) {
+                application.getFlinkClusterId()));
+        Map<String, String> config = cluster.getFlinkConfig();
+        if (!config.isEmpty()) {
+          savepointPath = config.get(CheckpointingOptions.SAVEPOINT_DIRECTORY.key());
+        }
+      } else {
+        // 3.2) At the yarn or k8s mode, then read the savepoint in flink-conf.yml in the bound
+        // flink
+        FlinkEnv flinkEnv = flinkEnvService.getById(application.getVersionId());
+        savepointPath =
+            flinkEnv.convertFlinkYamlAsMap().get(CheckpointingOptions.SAVEPOINT_DIRECTORY.key());
+      }
+    }
+
+    return savepointPath;
+  }
+
+  @Override
+  public void trigger(Long appId, @Nullable String savepointPath) {
+    log.info("Start to trigger savepoint for app {}", appId);
+    Application application = applicationService.getById(appId);
+
+    FlinkRESTAPIWatcher.addSavepoint(application.getId());
+
+    application.setOptionState(OptionState.SAVEPOINTING.getValue());
+    application.setOptionTime(new Date());
+    this.applicationService.updateById(application);
+
+    FlinkEnv flinkEnv = flinkEnvService.getById(application.getVersionId());
+
+    // infer savepoint
+    String customSavepoint = this.getFinalSavepointDir(savepointPath, application);
+
+    FlinkCluster cluster = flinkClusterService.getById(application.getFlinkClusterId());
+    String clusterId = getClusterId(application, cluster);
+
+    Map<String, Object> properties = this.tryGetRestProps(application, cluster);
+
+    TriggerSavepointRequest request =
+        new TriggerSavepointRequest(
+            flinkEnv.getFlinkVersion(),
+            ExecutionMode.of(application.getExecutionMode()),
+            clusterId,
+            application.getJobId(),
+            customSavepoint,
+            application.getK8sNamespace(),
+            properties);
+
+    CompletableFuture<SavepointResponse> savepointFuture =
+        CompletableFuture.supplyAsync(() -> FlinkClient.triggerSavepoint(request), executorService);
+
+    handleSavepointResponseFuture(application, savepointFuture);
+  }
+
+  private void handleSavepointResponseFuture(
+      Application application, CompletableFuture<SavepointResponse> savepointFuture) {
+    CompletableFutureUtils.runTimeout(
+            savepointFuture,
+            10L,
+            TimeUnit.MINUTES,
+            savepointResponse -> {
+              if (savepointResponse != null && savepointResponse.savePointDir() != null) {
+                String savePointDir = savepointResponse.savePointDir();
+                log.info("Request savepoint successful, savepointDir: {}", savePointDir);
+              }
+            },
+            e -> {
+              log.error("Trigger savepoint for flink job failed.", e);
+              ApplicationLog log = new ApplicationLog();
+              log.setAppId(application.getId());
+              log.setYarnAppId(application.getClusterId());
```

Comment:
yarn appid? It seems inaccurate, if execmode is yarn, it needs to be set, but k8s mode is not